OpenRA.Game/Traits/Player/FrozenActorLayer.cs

```diff
@@ -129,7 +138,7 @@ public IEnumerable<IRenderable> Render(WorldRenderer wr)
 			return renderables;
 		}
 
-		public bool HasRenderables { get { return renderables.Any(); } }
+		public bool HasRenderables { get { return !Shrouded && renderables.Any(); } }
```

Comment:
This check shouldn't be here. It should be up to the caller. i.e. UnitOrderGenerator should make the check (since it's already checking both Visible and HasRenderables anyway).